app/controllers/api/public/v1/application_controller.rb

```diff
@@ -2,6 +2,11 @@ module API
   module Public
     module V1
       class ApplicationController < ActionController::API
+        rescue_from ActiveRecord::RecordNotFound, with: :jsonapi_404
```

Comment:
I think we need a standardised way of dealing with these types of errors. What about something like:

```
    def render_json_error(code: nil, message:, status: :internal_server_error)
      render json: { code: code, error: message }.compact, status: status
    end
```

Reply:
I've added a card for this to my ideas bored. We can chat about it